sql/parser/builtins.go

```diff
@@ -943,6 +961,94 @@ func datumToString(datum Datum) (string, error) {
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
 
+func regexpReplace(s, pattern, to, sqlFlags string) (Datum, error) {
+	pattern, global, err := regexpEvalFlags(pattern, sqlFlags)
+	if err != nil {
+		return nil, err
+	}
+
+	patternRe, err := regexp.Compile(pattern)
+	if err != nil {
+		return nil, err
+	}
+
+	matchCount := 1
+	if global {
+		matchCount = -1
+	}
+
+	replaceIndex := 0
+	var newString bytes.Buffer
+	for _, matchIndex := range patternRe.FindAllStringSubmatchIndex(s, matchCount) {
+		start := matchIndex[0]
+		end := matchIndex[1]
+
+		preMatch := s[replaceIndex:start]
+		newString.WriteString(preMatch)
+
+		matchTo := to
```

Comment:
I reworked a lot of this replacement expansion because the old way could lead to incorrect expansion chaining (ie. \1 return "a\2b", which would cause \2 to be expanded)